### PR TITLE
Package VisIt: updated visit package

### PR DESCRIPTION
**Pull request by dorier**
This PR updates the VisIt package with the following additions:

- HDF5 is now an option that can be disabled
- SILO is now an option that can be disabled
- Python is now an option that can be disabled
- GUI is now an option that can be disabled
- Can enable "parallel" to compile the parallel engine

This allows to better select the needed dependencies and in particular to compile just the engine by disabling the gui, hdf5, and silo variants. This is useful when one wants to compile just the libsim in situ visualization interface (i.e. just the engine).

### CODE REVIEW

**Comment by dorier** [comment]
@alexmyczko  you might have problems with the VTK package; I have submitted an issue here https://github.com/spack/spack/issues/9423 and a PR to solve if here https://github.com/spack/spack/pull/9431

**Comment by alexmyczko** [comment]
I already was having an issue with spack install visit (without your pr), that spack would complain multiple gl providers. I'm on Ubuntu 18.04

**Comment by dorier** [comment]
@alexmyczko For me (on debian) I install it using `spack install visit ^vtk~opengl2+osmesa`. This seems to get rid of the gl provider problem.

**Comment by dorier** [comment]
One of the tests didn't pass, apparently because of a timeout in a connection with Codecov. Not related to this package in particular. Is it possible to re-run it?

**Comment by alexmyczko** [comment]
It failed at the end with: 

==> No patches needed for openmpi
==> Building openmpi [AutotoolsPackage]
==> Executing phase: 'autoreconf'
==> Error: InstallError: OpenMPI requires both C and Fortran compilers!

/opt/spack/var/spack/repos/builtin/packages/openmpi/package.py:347, in die_without_fortran:
        344        # avoid delayed build errors in dependents.
        345        if (self.compiler.f77 is None) or (self.compiler.fc is None):        346            raise InstallError(
  >>    347                'OpenMPI requires both C and Fortran compilers!'
        348            )

See build log for details:
  /opt/spack/var/spack/stage/openmpi-3.1.2-txgdzuhtcyxgfokaagg44l6xmkqyibwn/openmpi-3.1.2/spack-build.out

**Comment by dorier** [comment]
That's still unrelated to this specific package. It seems to be the underlying test setup.

**Comment by dorier** [comment]
@alexmyczko were you talking about the tests or about your own attempt to install the package?

If it's the latter, the error you are getting is because this VisIt package has a `+parallel` variant enabled by default. This variant requires MPI, which will trigger the installation of either Mpich or OpenMPI on your platform. However both Mpich and OpenMPI require a compiler that support Fortran (which for instance is not the case of clang).

You can try again with the `~parallel` option.

File: var/spack/repos/builtin/packages/visit/package.py [hunk 1]
```diff
@@ -38,33 +38,65 @@ class Visit(CMakePackage):
     version('2.10.2', '253de0837a9d69fb689befc98ea4d068')
     version('2.10.1', '3cbca162fdb0249f17c4456605c4211e')
 
+    variant('gui',    default=True, description="Enable VisIt's GUI")
```
Comment (ax3l) [inline var/spack/repos/builtin/packages/visit/package.py]: there are little inconsistent quotes in description argument :)

File: var/spack/repos/builtin/packages/visit/package.py [hunk 2]
```diff
@@ -38,33 +38,65 @@ class Visit(CMakePackage):
...
+    variant('hdf5',   default=True, description='Enable HDF5 file format')
+    variant('silo',   default=True, description='Enable Silo file format')
+    variant('python', default=True, description='Enable Python support')
+    variant('parallel', default=False, description='Enable parallel engine')
```
Comment (ax3l) [inline var/spack/repos/builtin/packages/visit/package.py]: we usually call this variant `mpi` as well in spack and make it `default=True`: https://spack.readthedocs.io/en/latest/packaging_guide.html#variant-names

**Comment by ax3l** [comment]
@alexmyczko currently, you need a Fortran compiler in your `compilers.yaml` in order to build OpenMPI:
https://spack.readthedocs.io/en/latest/getting_started.html#compiler-configuration

**Comment by ax3l** [comment]
@alexmyczko does this package work for you? From a pure code review, it looks good to me.

**Comment by dorier** [comment]
Any chance to get this PR in soon? I'd like to make another package depend on it :-) Thanks

**Pull request merged**